modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt

```diff
@@ -2226,11 +2228,34 @@ open class PlaybackManager @Inject constructor(
                 if (isPlaying()) {
                     statsManager.addTotalListeningTime(UPDATE_TIMER_POLL_TIME)
                 }
+                if (playbackStateRelay.blockingFirst().isSleepTimerRunning && !sleepAfterEpisode) { // Does not apply to end of episode sleep time
+                    setupFadeOutWhenFinishingSleepTimer()
+                }
             }
             .switchMapCompletable { updateCurrentPositionRx() }
             .subscribeBy(onError = { Timber.e(it) })
     }
 
+    private fun setupFadeOutWhenFinishingSleepTimer() {
+        // Needs to run in main thread because of player getVolume
+        Handler(Looper.getMainLooper()).post {
```

Comment:
Can you change it from posting to a Handler to launching a coroutine on the main thread? Launching a coroutine will respect scope's lifecycle.

Reply:
Sure

Reply:
Here: [a3b91a4](https://github.com/Automattic/pocket-casts-android/pull/2069/commits/a3b91a4c2a134007c60ed72f25544b585362c51c)